src/sentry/api/endpoints/integrations/sentry_apps/requests.py

```diff
@@ -11,15 +13,21 @@
 class SentryAppRequestsEndpoint(SentryAppBaseEndpoint):
     permission_classes = (SentryAppStatsPermission,)
 
-    def format_request(self, request: Request, sentry_app):
+    def format_request(self, request: Request, sentry_app, org_slug: str = None):
```

Comment:
I think this would make more sense as a serializer class:
```py
from __future__ import annotations
from typing import Any, Mapping, MutableMapping

from django.urls import reverse

from sentry import eventstore
from sentry.api.serializers import Serializer
from sentry.models import Organization, Project, SentryApp


class RequestSerializer(Serializer):
    def __init__(self, sentry_app: SentryApp) -> None:
        self.sentry_app = sentry_app

    def get_attrs(self, item_list: list[Any], user: Any, **kwargs: Any) -> MutableMapping[Any, Any]:
        project_ids = {item["project_id"] for item in item_list}
        projects = Project.objects.filter(id__in=project_ids)
        projects_by_id = {project.id: project for project in projects}

        organization_ids = {item["organization_id"] for item in item_list}
        organizations = Organization.objects.filter(id__in=organization_ids)
        organizations_by_id = {organization.id: organization for organization in organizations}

        return {
            item: {
                "organization": organizations_by_id.get(item["organization_id"]),
                "project": projects_by_id.get(item["project_id"]),
            }
            for item in item_list
        }

    def serialize(
        self,
        obj: Any,
        attrs: Mapping[Any, Any],
        user: Any,
        **kwargs: Any
    ) -> Mapping[str, Any]:
        organization = attrs.get("organization")
        project = attrs.get("project")
        response_code = obj.get("response_code")

        data = {
            "webhookUrl": obj.get("webhook_url"),
            "sentryAppSlug": self.sentry_app.slug,
            "eventType": obj.get("event_type"),
            "date": obj.get("date"),
            "responseCode": response_code,
        }

        if response_code >= 400 or response_code == 0:
            data.update({
                "requestBody": obj.get("request_body"),
                "requestHeaders": obj.get("request_headers"),
                "responseBody": obj.get("response_body"),
            })

        if project and "error_id" in obj:
            # Make sure the project actually belongs to the org that owns the Sentry App
            if project.organization_id == self.sentry_app.owner_id:
                # Make sure the event actually exists
                event = eventstore.get_event_by_id(project.id, obj["error_id"])
                if event is not None and event.group_id is not None:
                    data["errorUrl"] = reverse(
                        "sentry-organization-event-detail",
                        args=[project.organization.slug, event.group_id, event.event_id],
                    )

        if organization:
            data["organization"] = {"name": organization.name, "slug": organization.slug}

        return data
```

Reply:
@mgaeta why is `user` passed in if it's not used?